lib/src/lib.rs

```diff
@@ -665,28 +640,153 @@ impl Repo {
             let seckey_path = sec_key_file_path(&repo_path);
             write_seckey_file(&seckey_path, &encrypted_seckey, &nonce, &salt)?;
         }
-        {
-            let version_path = version_file_path(&repo_path);
-            let mut seckey_file = try!(fs::File::create(version_path));
-            let mut writer = &mut seckey_file as &mut Write;
-            try!(write!(writer, "{}", REPO_VERSION_CURRENT));
+        Repo::write_version_file(&repo_path, 0)?;
+
+        Ok(())
+    }
+
+    fn write_config_v1(repo_path: &Path, pk: &box_::PublicKey, sk: &box_::SecretKey, passphrase : &str) -> Result<()> {
```

Comment:
Yeah if you can then it makes lib.rs just the core logic and glue for the other modules.